src/components/token/TokenBalances.js

```diff
@@ -0,0 +1,327 @@
+     *                  We use this array to store already-calculated values,
+     *                  so we do not need to recalculate them if user is requesting an already-navigated page.
+     * loading: Initial loading, when user clicks on the Tokens navigation item
+     * isSearchLoading: Indicates if search results are being retrieved from explorer-service
+     * calculatingPage: Indicates if next page is being retrieved from explorer-service
+     * error: Indicates if an unexpected error happened when calling the explorer-service
+     * tokenBalanceInformationError: Indicates if an unexpected error happened when calling the token balance information service 
+     * maintenanceMode: Indicates if explorer-service or its downstream services are experiencing problems. If so, maintenance mode is enabled as
+     *                  a "circuit breaker" to remove additional load until the team fixes the problem
+     */
+    this.state = {
+      tokenId: '00',
+      tokenBalances: [],
+      hasAfter: false,
+      hasBefore: false,
+      sortBy: 'total',
+      order: 'desc',
+      page: 1,
+      pageSearchAfter: [],
+      loading: false,
+      isSearchLoading: false,
+      calculatingPage: false,
+      error: false,
+      tokenBalanceInformationError: false,
+      maintenanceMode: this.props.maintenanceMode,
+      transactionsCount: 0,
+      addressesCount: 0,
+    };
+  }
+
+  componentDidMount = async () => {
+    if (this.state.maintenanceMode) {
+      return;
+    }
+
+    // 'Click' on search to make the first query
+    const queryParams = this.pagination.obtainQueryParams();
+
+    await this.setState({
+      sortBy: get(queryParams, 'sortBy', this.state.sortBy),
+      order: get(queryParams, 'order', this.state.order),
+      loading: true,
+    });
+
+    await this.performSearch();
+
+    this.setState({
+      loading: false,
+    });
+  }
+
+  /**
+   * Call explorer-service to get list of token balances for a given tokenId
+   *
+   * @param {*} searchAfter Parameter needed by ElasticSearch for pagination purposes
+   * @returns tokens
+   */
+  getTokenBalances = async (searchAfter) => {
+    const tokenBalancesRequest = await tokensApi.getBalances(this.state.tokenId, this.state.sortBy, this.state.order, searchAfter);
+    this.setState({
+      error: get(tokenBalancesRequest, 'error', false),
+    });
+    return get(tokenBalancesRequest, 'data', { hits: [], 'has_next': false });
+  }
+
+  loadTokenBalanceInformation = async () => {
+    const tokenBalanceInformationRequest = await tokensApi.getBalanceInformation(this.state.tokenId);
+    this.setState({
+      tokenBalanceInformationError: get(tokenBalanceInformationRequest, 'error', false),
+    });
+
+    return get(tokenBalanceInformationRequest, 'data', {
+      transactions: 0,
+      addresses: 0,
+    });
+  }
+
+  /**
+   * Calls elastic search with state data, sets loading and URL information
```

Comment:
I think we need to explain that ElasticSearch is not called directly, but with explorer-service as gateway

Reply:
Fixed! Thanks